### PR TITLE
Feature/blob disk support

### PR DESCRIPTION
**Pull request by PoisonousJohn**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [X] Have you run `./scripts/ci/test_static.sh` locally? (`pip install pylint flake8` required)
- [X] Have you run `python scripts/ci/test_index.py -q` locally?

Added support for images based on Blob disks.
I tested on an older version of extension, which is installed via az cli, and it worked.

Could you please review & test it?

This PR addresses #84.

### CODE REVIEW

File: src/image-copy/azext_imagecopy/custom.py [hunk 1]
```diff
@@ -127,13 +153,11 @@ def create_resource_group(resource_group_name, location):
...
 
-    if 'true' in cmd_output:
-        return
-
-    # create the target resource group
-    logger.warn("Creating resource group: %s", resource_group_name)
-    cli_cmd = prepare_cli_command(['group', 'create',
-                                   '--name', resource_group_name,
-                                   '--location', location])
+    if 'false' in cmd_output:
```
Comment (tamirkamara) [inline src/image-copy/azext_imagecopy/custom.py]: Just being curios - why did you change this?

File: src/image-copy/azext_imagecopy/custom.py [hunk 2]
```diff
@@ -23,34 +25,57 @@ def imagecopy(source_resource_group_name, source_object_name, target_location,
...
 
-    source_os_disk_snapshot_url = json_output['accessSas']
-    logger.debug("source os disk snapshot url: %s",
-                 source_os_disk_snapshot_url)
+        json_output = run_cli_command(cli_cmd, return_as_json=True)
+
+        source_os_disk_snapshot_url = json_output['accessSas']
+        logger.debug("source os disk snapshot url: %s", source_os_disk_snapshot_url)
+    else:
+        logger.warn("blob OS disk detected")
```
Comment (tamirkamara) [inline src/image-copy/azext_imagecopy/custom.py]: logger.warn writes to the console, so I think you should change that and let the user know what is the action performed, something like: Getting sas url for the source blob.

File: src/image-copy/azext_imagecopy/custom.py [hunk 3]
```diff
@@ -23,34 +25,57 @@ def imagecopy(source_resource_group_name, source_object_name, target_location,
...
+        blob_uri = json_cmd_output['storageProfile']['osDisk']['blobUri'].replace('https://', '')
+        logger.warn("blob uri: %s" % blob_uri)
```
Comment (tamirkamara) [inline src/image-copy/azext_imagecopy/custom.py]: I believe this should be logger.debug or removed. If you decide to keep it I would also include the word "source" to distinct from the blob/snapshots used later on the "target" phase.

File: src/image-copy/azext_imagecopy/custom.py [hunk 4]
```diff
@@ -23,34 +25,57 @@ def imagecopy(source_resource_group_name, source_object_name, target_location,
...
+        blob_account = blob_uri.split('.', 1)[0]
+        blob_uri_components = blob_uri.split('/',)
+        blob_container = blob_uri_components[1]
+        blob_name = blob_uri_components[2]
+        cli_cmd = prepare_cli_command(['storage', 'blob', 'generate-sas',
+                                       '--https-only',
+                                       '--expiry',
+                                       (datetime.datetime.utcnow() +
```
Comment (tamirkamara) [inline src/image-copy/azext_imagecopy/custom.py]: I'd move the date to an external var (easier to read)

File: src/image-copy/azext_imagecopy/custom.py [hunk 5]
```diff
@@ -23,34 +25,57 @@ def imagecopy(source_resource_group_name, source_object_name, target_location,
...
+                                        datetime.timedelta(days=1))
+                                       .isoformat().split('.', 1)[0]+'Z',
+                                       '--permissions=r',
+                                       '--account-name', blob_account,
+                                       '--container-name', blob_container,
+                                       '-n', blob_name])
```
Comment (tamirkamara) [inline src/image-copy/azext_imagecopy/custom.py]: Please use full parameter names, i.e. --name (and not -n)

File: src/image-copy/azext_imagecopy/custom.py [hunk 6]
```diff
@@ -23,34 +25,57 @@ def imagecopy(source_resource_group_name, source_object_name, target_location,
...
+        sas = run_cli_command(cli_cmd, return_as_json=False).strip().replace('"', '')
+        full_sas_url = 'https://%s?%s' % (blob_uri, sas)
```
Comment (tamirkamara) [inline src/image-copy/azext_imagecopy/custom.py]: I think this should be merged with the following line

**Comment by PoisonousJohn** [comment]
@tamirkamara the job is failing due to `R: 18, 0: Too many local variables (29/25) (too-many-locals)`, which I didn't change.

File: src/image-copy/azext_imagecopy/custom.py (hunk 1 above)
Comment (PoisonousJohn) [reply src/image-copy/azext_imagecopy/custom.py]: That's my mistake. I applied a patch, based on older version of the script. So it's a regression I would say. I'll revert it back

**Comment by PoisonousJohn** [comment]
@tamirkamara thx for the review. Addressed all the changes you requested

**Comment by tamirkamara** [comment]
This PR addresses a use-case where an image is created from a vhd/blob disk = an un-managed disk. However, an image is managed by nature and it can create managed VMs only, so essentially this "converts" an un-managed disk VM to a managed one.

@PoisonousJohn and I agreed that it looks like this use case isn't very common and It's our believe that the extension should not support that.

@PoisonousJohn - Can you please close this PR?
